usmqe/api/tendrlapi/user.py

```diff
@@ -0,0 +1,142 @@
+"""
+tendrl REST API.
+"""
+
+import json
+import requests
+import pytest
+from usmqe.api.tendrlapi.common import TendrlApi
+
+LOGGER = pytest.get_logger("tendrlapi.user", module=True)
+
+
+class ApiUser(TendrlApi):
+    """ Main class for interact with REST API - user.
+    """
+    def users(self, asserts_in=None):
+        """ Get users.
+
+        Name:        "GET_users",
+        Method:      "GET",
+        Pattern:     "users",
+
+        Args:
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "users"
+        request = requests.get(
+            pytest.config.getini("usm_api_url") + pattern,
+            auth=self._auth)
+        self.print_req_info(request)
+        self.check_response(request, asserts_in)
+        if request.ok:
+            defined_keys = {
+                "email",
+                "username",
+                "name",
+                "role"}
+            for item in request.json(encoding='unicode'):
+                user = item["username"]
+                pytest.check(
+                    item.keys() == defined_keys,
+                    "User {0} should contain: {1}\n\tUser {0} contains: {2}"
+                    .format(user, defined_keys, item.keys))
+            return request.json(encoding='unicode')
+
+    def user_edit(self, username, data, asserts_in=None):
+        """ Edit a single user
+
+        Args:
+            username: name of user that is going to be updated
+            data: dictionary data about user
+                  have to contain: email, username, name, role
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "users/{}".format(username)
+        request = requests.put(
+            pytest.config.getini("usm_api_url") + pattern,
+            json.dumps(data),
+            auth=self._auth)
+        self.print_req_info(request)
+        self.check_response(request, asserts_in)
+# TODO check json comparision
+        return request.json(encoding='unicode')
+
+    def user_add(self, user_in, asserts_in=None):
+        """ Add user throught **users**.
+
+        Name:        "POST_users",
+        Method:      "POST",
+        Pattern:     "users",
+
+        Args:
+            user_in: dictionary info about user
+                     have to contain: name, username, email, role, password, password_confirmation
+            asserts_in: assert values for this call and this method
+        """
+        pattern = "users"
+        request = requests.post(
+            pytest.config.getini("usm_api_url") + pattern,
+            data=json.dumps(user_in),
+            auth=self._auth)
+        self.print_req_info(request)
+        self.check_response(request, asserts_in)
+        user_in.pop("password", None)
+        user_in.pop("password_confirmation", None)
```

Comment:
I put it there so that the password is not obtainable from response of this function and if the password is gone you can compare dictionary from this function to response from GET `hostname/api/1.0/users/someusername`. We are currently not doing this comparison anywhere so this code could be gone.